tests/python/test_ad_for_fwd.py

```diff
@@ -0,0 +1,73 @@
+import taichi as ti
+from tests import test_utils
+
+
+@test_utils.test()
+def test_ad_sum_fwd():
+    N = 10
+    a = ti.field(ti.f32, shape=N)
+    b = ti.field(ti.i32, shape=N)
+    p = ti.field(ti.f32, shape=N)
+    ti.root.lazy_dual()
+
+    @ti.kernel
+    def compute_sum():
+        for i in range(N):
+            ret = 1.0
+            for j in range(b[i]):
+                ret = ret + a[i]
+            p[i] = ret
+
+    for i in range(N):
+        a[i] = 3
+        b[i] = i
+
+    compute_sum()
+
+    for i in range(N):
+        assert p[i] == 3 * b[i] + 1
+
+    with ti.ad.FwdMode(loss=p, parameters=a, seed=[1.0 for _ in range(N)]):
+        compute_sum()
+
+    for i in range(N):
+        assert p.dual[i] == b[i]
+
```

Comment:
shall we add another FwdMode to test agaist clear_dual and non-all-one seeds? ;)